docs/reference/modules/coordination.asciidoc

```diff
@@ -0,0 +1,497 @@
+situation if using a more automated orchestrator, particularly if a network
+partition happens at the wrong time.
+
+We avoid this by requiring a separate _cluster bootstrapping_ process to take
+place on every brand-new cluster. This is only required the very first time the
+whole cluster starts up: new nodes joining an established cluster can safely
+obtain all the information they need from the elected master, and nodes that
+have previously been part of a cluster will have stored to disk all the
+information required when restarting.
+
+A cluster can be bootstrapped by sending a _bootstrap warrant_ to any of its
+master-eligible nodes.  A bootstrap warrant is a document that contains the
+information that the cluster needs to finish forming, including the identities
+of the master-eligible nodes that form its first voting configuration, and
+looks like this:
+
+[source,js]
+--------------------------------------------------
+{
+  "master_nodes":[
+    {"id":"USpTGYaBSIKbgSUJR2Z9lg","name":"master-a"},
+    {"id":"gSUJR2Z9lgUSpTGYaBSIKb","name":"master-b"},
+    {"id":"2Z9lgUSpTgSUYaBSIKbJRG","name":"master-c"}
+  ]
+}
+--------------------------------------------------
+
+To bootstrap a cluster, the administrator must identify a suitable set of
+master-eligible nodes, construct a bootstrap warrant, and pass the warrant to
+the `POST /_cluster/bootstrap` API:
+
+[source,js]
+--------------------------------------------------
+# send the bootstrap warrant back to the cluster
+POST /_cluster/bootstrap
+{
+  "master_nodes":[
+    {"id":"USpTGYaBSIKbgSUJR2Z9lg","name":"master-a"},
+    {"id":"gSUJR2Z9lgUSpTGYaBSIKb","name":"master-b"},
+    {"id":"2Z9lgUSpTgSUYaBSIKbJRG","name":"master-c"}
+  ]
+}
+--------------------------------------------------
+// CONSOLE
+
+This only needs to occur once, on a single master-eligible node in the cluster,
+but for robustness it is safe to repeatedly call `POST /_cluster/bootstrap`,
+and to call it on different nodes concurrently. However **it is vitally
+important** to use the same bootstrap warrant in each call.
+
+WARNING: You must pass the same bootstrap warrant to each call to `POST
+/_cluster/bootstrap` in order to be sure that only a single cluster forms
+during bootstrapping and therefore to avoid the risk of data loss.
+
+The simplest and safest way to construct a bootstrap warrant is to use the `GET
+/_cluster/bootstrap` API:
+
+[source,js]
+--------------------------------------------------
+# Immediately return a bootstrap warrant based on the nodes discovered so far
+GET /_cluster/bootstrap
+# Wait until the node has discovered at least 3 nodes, or 60 seconds has elapsed,
+# and then return the resulting bootstrap warrant
+GET /_cluster/bootstrap?wait_for_nodes=3&timeout=60s
+--------------------------------------------------
+// CONSOLE
+
+This API returns a properly-constructed bootstrap warrant that is ready to pass
+to the `POST /_cluster/bootstrap` API.  It includes all of the master-eligible
```

Comment:
an extra space 🗡

Reply:
No idea how to get `vi` to avoid doing that when reflowing paragraphs. It's probably in more places too.